src/term.rs

```diff
@@ -836,17 +891,34 @@ pub enum UnaryOp {
     /// Transform a string to an enum.
     EnumFromStr(),
     /// Test if a regex matches a string.
-    /// Like [`StrMatch`], this is a unary operator because we would like a way to share the
+    /// Like [`UnaryOp::StrMatch`], this is a unary operator because we would like a way to share the
     /// same "compiled regex" for many matching calls. This is done by returning functions
-    /// wrapping [`StrIsMatchCompiled`] and [`StrMatchCompiled`]
+    /// wrapping [`UnaryOp::StrIsMatchCompiled`] and [`UnaryOp::StrMatchCompiled`]
     StrIsMatch(),
     /// Match a regex on a string, and returns the captured groups together, the index of the
     /// match, etc.
     StrMatch(),
-    /// Version of `StrIsMatch` which remembers the compiled regex.
+    /// Version of [`UnaryOp::StrIsMatch`] which remembers the compiled regex.
     StrIsMatchCompiled(CompiledRegex),
-    /// Version of `StrMatch` which remembers the compiled regex.
+    /// Version of [`UnaryOp::StrMatch`] which remembers the compiled regex.
     StrMatchCompiled(CompiledRegex),
+    /// Force "full" evaluation of a term and return it.
+    ///
+    /// This was added in the context of [`BinaryOp::ArrayLazyAssume`],
+    /// and may not make much sense on its own.
```

Comment:
```suggestion
    /// This was added in the context of [`BinaryOp::ArrayLazyAssume`],
    /// in particular to make serialization work with lazy array contracts.
```

I'm not sure it doesn't make sense. It may make more sense than the current `deep_seq`, somehow (it evaluates more). If I may make a suggestion, we could stick to the motivation :slightly_smiling_face: